src/coreclr/jit/lowerxarch.cpp

```diff
@@ -7511,6 +7553,154 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* parentNode, GenTre
     }
 }
 
+//----------------------------------------------------------------------------------------------
+// MakeHWIntrInsicSrcContained: Unfold the eligible constant vector when embedded broadcast is
+//                              available.
+//
+//  Arguments:
+//     parentNode - The hardware intrinsic node
+//     childNode  - The operand node to try contain
+//
+void Lowering::MakeHWIntrinsicSrcContained(GenTreeHWIntrinsic* parentNode, GenTree* childNode)
+{
+    assert(childNode->OperIs(GT_CNS_VEC));
```

Comment:
It might be better to just make this do something like.
```cpp
if (childNode->IsVectorConst() && parentNode->OperIsEmbBroadcastCompatible())
{
    childNode = TryFoldCnsVecForEmbeddedBroadcast(parentNode, childNode->AsVecCon());
}
MakeSrcContained(parentNode, childNode);
```

That makes it a small and generally reusable method for any HWIntrinsic containment and isolates the folding to a very clear function. How it is now, the function looks general purpose from the name, but actually only works for a very specific edge case.